app/src/main/java/ca/carleton/gcrc/n2android_mobile1/connection/ConnectionSyncProcess.java

```diff
@@ -152,4 +242,99 @@ public boolean updateLocalSkeletonDocument(JSONObject doc) throws Exception {
             throw new Exception("Unable to update local skeleton documents",e);
         }
     }
+
+    public void updateRemoteDocuments(List<JSONObject> remoteDocuments) throws Exception {
+        try {
+            List<JSONObject> newLocalDocuments = getNewLocalDocuments(remoteDocuments);
+
+            for(JSONObject doc : newLocalDocuments) {
+                updateServerDocument(doc);
+            }
+
+        } catch (Exception e) {
+            throw new Exception("Unable to update remote documents",e);
+        }
+    }
+
+    public List<JSONObject> getNewLocalDocuments(List<JSONObject> remoteDocuments) throws Exception {
+        try {
+            List<JSONObject> localDocuments = documentDb.getAllDocuments();
+            List<JSONObject> documentsMissingFromRemote = new ArrayList<JSONObject>();
+
+            Set<String> localDocumentsSet = new HashSet<String>();
+
+            for (JSONObject doc : localDocuments) {
+                localDocumentsSet.add (doc.getString("_id"));
+            }
+
+            for (JSONObject doc : remoteDocuments) {
+                if (localDocumentsSet.contains(doc.getString("_id"))) {
+                    localDocumentsSet.remove(doc.getString("_id"));
```

Comment:
small optimization: we could store doc.getString() in a variable to avoid fetching it twice in every loop